homeassistant/components/brother/config_flow.py

```diff
@@ -64,6 +64,85 @@ class BrotherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        try:
+            if "Brother" not in user_input["name"]:
+                raise ValueError
+        except (KeyError, ValueError):
+            return self.async_abort(reason="not_brother_printer")
+
+        # Hostname is format: brother.local.
+        host = user_input["hostname"].rstrip(".")
+
+        try:
+            brother = Brother(host)
+            await brother.async_update()
+        except (ConnectionError, SnmpError, UnsupportedModel):
+            return self.async_abort(reason="connection_error")
+
+        # Check if already configured
+        await self.async_set_unique_id(brother.serial.lower())
+        self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update(
+            {
+                CONF_HOST: host,
+                "title_placeholders": {
+                    "serial_number": brother.serial,
+                    "model": brother.model,
+                },
+            }
+        )
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
```

Comment:
Don't render forms for other steps. Instead `return await self.async_step_zeroconf_confirm()`